tensorflow/contrib/gdr/gdr_memory_manager.cc

```diff
@@ -0,0 +1,608 @@
+#ifdef TENSORFLOW_USE_GDR
+
+#include "tensorflow/contrib/gdr/gdr_memory_manager.h"
+
+#include <atomic>
+#include <cerrno>
+#include <fstream>
+#include <list>
+#include <map>
+
+#include <fcntl.h>
+#include <rdma/rdma_cma.h>
+#include <rdma/rdma_verbs.h>
+#include <sys/epoll.h>
+
+#include "tensorflow/contrib/gdr/gdr.pb.h"
+#include "tensorflow/core/common_runtime/bfc_allocator.h"
+#include "tensorflow/core/common_runtime/device.h"
+#include "tensorflow/core/common_runtime/dma_helper.h"
+#if GOOGLE_CUDA
+#include "tensorflow/core/common_runtime/gpu/gpu_util.h"
+#include "tensorflow/core/common_runtime/gpu/process_state.h"
+#endif  // GOOGLE_CUDA
+#include "tensorflow/core/framework/allocator_registry.h"
+#include "tensorflow/core/lib/core/status.h"
+#include "tensorflow/core/platform/macros.h"
+#include "tensorflow/core/platform/mutex.h"
+
+namespace tensorflow {
+
+namespace {
+
+bool IsGDRAvailable() {
+  std::ifstream ifs("/proc/modules");
```

Comment:
There are 3 Linux specific dependencies:

1. the [RDMA user space libraries](https://github.com/linux-rdma/rdma-core), co-evolving with all [kernel drivers](https://github.com/torvalds/linux/tree/master/drivers/infiniband) in Linux source tree
2. the [nv_peer_mem](https://github.com/Mellanox/nv_peer_memory) Linux kernel module, developed by Menallox and NVIDIA using their [kernel APIs](http://docs.nvidia.com/cuda/gpudirect-rdma/index.html#kernel-api)
3. detection of numa node of IB device using `/proc`, as similar to [what is being done for CUDA device](https://github.com/tensorflow/tensorflow/blob/45254d9d7f790f97746775967637773d137682a9/tensorflow/stream_executor/cuda/cuda_gpu_executor.cc#L899)

Being said that, I agree with you that a partial fix could be introduced in the same way as it is handled in #5991.